test/system/alpha/tooltip_test.rb

```diff
@@ -8,8 +8,94 @@ def test_renders
+
+      find("button").send_keys("") # sends focus to button
+
+      assert_selector("tool-tip.position-absolute", visible: true)
+      refute_selector("tool-tip.position-absolute.sr-only", visible: true)
+      style = find("tool-tip", visible: true)["style"]
+      assert_equal(true, style.include?("top"))
+      assert_equal(true, style.include?("left"))
+    end
+
+    def test_appears_and_positions_on_hover_of_button
+      visit_preview(:default)
+
+      assert_nil(find("tool-tip", visible: :hidden)["style"]) # position not set on initial load
+      assert_selector("tool-tip.position-absolute.sr-only", visible: :hidden)
+
+      find("button").hover
+
+      assert_selector("tool-tip.position-absolute", visible: true)
+      refute_selector("tool-tip.position-absolute.sr-only", visible: true)
+      style = find("tool-tip", visible: true)["style"]
+      assert_equal(true, style.include?("top"))
+      assert_equal(true, style.include?("left"))
+    end
+
+    def test_hides_tooltip_on_escape
+      visit_preview(:default)
+
+      find("button").send_keys("") # focus
+      assert_selector("tool-tip", visible: true)
+
+      find("button").send_keys(:escape)
+      assert_selector("tool-tip", visible: :hidden)
+    end
+
+    def test_appends_to_existing_aria_labelledby
+      visit_preview(:label_tooltip_on_button_with_existing_labelledby)
+
+      tooltip_id = find("tool-tip", visible: :hidden)["id"]
+      assert_equal("existing-label-id #{tooltip_id}", find("button")["aria-labelledby"])
+    end
+
+    def test_appends_to_existing_aria_describedby
+      visit_preview(:description_tooltip_on_button_with_existing_describedby)
+
+      tooltip_id = find("tool-tip", visible: :hidden)["id"]
+      assert_equal("existing-description-id #{tooltip_id}", find("button")["aria-describedby"])
+    end
+
+    def test_always_aria_hidden_when_tooltip_is_label
+      visit_preview(:label_tooltip_on_button_with_existing_labelledby)
+
+      assert_selector("tool-tip[aria-hidden='true']", visible: :hidden)
+      find("button").send_keys("") # focus
+      assert_selector("tool-tip[aria-hidden='true']", visible: :visible)
+    end
+
+    def test_never_aria_hidden_when_tooltip_is_description
+      visit_preview(:default)
+
+      assert_selector("tool-tip.sr-only", visible: :hidden)
+      refute_selector("tool-tip.sr-only[aria-hidden]", visible: :hidden)
+
+      find("button").send_keys("") # focus
+
+      refute_selector("tool-tip.sr-only", visible: :hidden)
+      assert_selector("tool-tip", visible: :visible)
+      refute_selector("tool-tip[aria-hidden]", visible: :visible)
+    end
+
+    def test_does_not_overflow_and_affect_layout
+      visit_preview(:with_right_most_position)
+      tooltip_in_viewport = evaluate_script("(function(el) {
+        return (
+          el.getBoundingClientRect().right <= (window.innerWidth || document.documentElement.clientWidth) + 1
```

Comment:
In this PVC preview environment, the button control is at the right-most side of the page and has no padding. The sr-only will give the tooltip a width of 1px causing a 1px overflow in this preview environment.

In practice however, all GitHub controls contained within a UI will have a padding of at least a few pixels so a tooltip with 1px should not cause any horizontal overflow.

I verified this 1px doesn't cause any problems in https://github.com/primer/view_components/pull/1307#issuecomment-1215997460.

(chatted about this with @langermank)